docs/website/docs/tutorial/filesystem.md

```diff
@@ -0,0 +1,276 @@
+- `requirements.txt`: This file lists all the Python dependencies required for your project.
+- `.dlt/`: This directory contains the [configuration files](../general-usage/credentials/) for your project:
+    - `secrets.toml`: This file stores your API keys, tokens, and other sensitive information.
+    - `config.toml`: This file contains the configuration settings for your dlt project.
+
+:::note
+When deploying your pipeline in a production environment, managing all configurations with files
+might not be convenient. In this case, you can use environment variables or other [configuration providers](../general-usage/credentials/setup#available-config-providers) available in
+`dlt` to store secrets and configs instead.
+:::
+
+## 2. Creating the pipeline
+
+Filesystem source provides users with building blocks for loading data from any type of files. You can break down the data extraction into two steps:
+
+1. Accessing files in the bucket / directory.
+2. Reading the files and yielding records.
+
+`dlt`'s filesystem source includes several resources that you can use together or individually:
+
+- `filesystem` resource accesses files in the directory or bucket
+- several readers resources (`read_csv`, `read_parquet`, `read_jsonl`) read files and yield the records
+
+Let's initialize a source and create a pipeline for loading parquet files from an AWS S3 to DuckDB:
+
+```
+import dlt
+from dlt.sources.filesystem import filesystem, read_parquet
+
+parquet_resource = filesystem(file_glob="**/*.parquet") | read_parquet()
+pipeline = dlt.pipeline(pipeline_name="s3_to_duckdb", dataset_name="parquet_data", destination="duckdb")
+info = pipeline.run(parquet_resource)
+print(info)
+```
+
+You can replace the example code from `filesystem_pipeline.py` with the code above. What's happening in this snippet?
+
+1. We initialize the filesystem resource and pass the `file_glob` parameter. Based in this parameter, `dlt` will filter
+all files in the bucket.
+2. We pipe the transformer `read_parquet` to read the files yielded by filesystem resource and iterate over records
+from the file.
+
+  ```note
+  A **transformer** in `dlt` is a special type of resource that processes each record from another resource. This lets you
+  chain multiple resources together. To learn more, check out the
+  ["@dlt.transformer section"](../general-usage/resource#process-resources-with-dlttransformer).
+  ```
+
+3. We create the `dlt` pipeline with the name `s3_to_duckdb` and DuckDB destination and run this pipeline.
+
+## 3. Configuring filesystem source
+
+Next, we need to configure the connection. Specifically, we’ll set the bucket URL and AWS credentials.
+This example focuses on AWS S3. For other Cloud Storage services, see the [Filesystem configuration section](../dlt-ecosystem/verified-sources/filesystem/basic#configuration).
+
+Let's specify the bucket url and AWS credentials:
```

Comment:
are you sure we should include all three options of specifying the credentials? In the earlier example we only have the toml example